### ISSUE TITLE
Gradient accumulation not properly implemented

### ISSUE THREAD

**Issue by clemsgrs** [mahmoodlab/HIPT#53]
Hi, based on the following lines, it seems gradient accumulation is not properly implemented:

https://github.com/mahmoodlab/HIPT/blob/a9b5bb8d159684fc4c2c497d68950ab915caeb7e/2-Weakly-Supervised-Subtyping/utils/core_utils.py#L285-L290

A proper implementation should look like the following:

```python
if gc > 1:
      loss = loss / gc

loss.backward()

if gc > 1 and (batch_idx + 1) % gc == 0:
      optimizer.step()
      optimizer.zero_grad()
```

**Comment by vildesboe** [comment]
> Hi, based on the following lines, it seems gradient accumulation is not properly implemented:
> 
> https://github.com/mahmoodlab/HIPT/blob/a9b5bb8d159684fc4c2c497d68950ab915caeb7e/2-Weakly-Supervised-Subtyping/utils/core_utils.py#L285-L290
> 
> A proper implementation should look like the following:
> 
> ```python
> loss = loss / gc
> loss.backward()
> 
> if (batch_idx + 1) % gc == 0:
>       optimizer.step()
>       optimizer.zero_grad()
> ```

Hi! I'm also working on reproducing this HIPT paper. Would you be interested in some discussion?

**Comment by clemsgrs** [comment]
sure, happy to chat. 
I’ve made my own version of the code here: https://github.com/clemsgrs/hipt

you can contact me at: clement (dot) grisi (at) radboudumc (dot) nl
